client/src/graphics/voxels/mod.rs

```diff
@@ -146,27 +146,24 @@ impl Voxels {
             }
 
             use Chunk::*;
-            for chunk in Vertex::iter() {
+            for vertex in Vertex::iter() {
+                let chunk = ChunkId::new(node, vertex);
                 // Fetch existing chunk, or extract surface of new chunk
                 match sim
                     .graph
-                    .get_mut(node)
-                    .as_mut()
+                    .get_chunk_mut(chunk)
                     .expect("all nodes must be populated before rendering")
-                    .chunks[chunk]
                 {
                     Generating => continue,
                     Fresh => {
                         // Generate voxel data
                         if let Some(params) = common::worldgen::ChunkParams::new(
                             self.surfaces.dimension() as u8,
                             &sim.graph,
-                            node,
                             chunk,
                         ) {
                             if self.worldgen.load(ChunkDesc { node, params }).is_ok() {
-                                sim.graph.get_mut(node).as_mut().unwrap().chunks[chunk] =
-                                    Generating;
+                                *sim.graph.get_chunk_mut(chunk).unwrap() = Generating;
```

Comment:
nit: Perhaps we should `impl Index(Mut)<ChunkId>` for panicing assets to save writing `unwrap` a bunch.

Reply:
That seems like a good idea. There is the risk of reduced mindfulness when making `unwrap` implicit. However, this tradeoff has already been considered in Rust standard library, and they decided in favor of allowing this, so I am willing to believe the benefits outweigh the costs.